### PR TITLE
fix vault client bundling

### PR DESCRIPTION
**Pull request by ori-shalom**
Fix bundling of vault client:
- Replace axios with fetch to reduce bundle size
- Allow import as both esm and commonjs

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,12 @@
 {
   "compilerOptions": {
-    "target": "ES2016",                       /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
-    "module": "commonjs",                     /* Specify what module code is generated. */
+    "target": "ES2015",                       /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
```
Comment (nirhaas) [inline tsconfig.json:3]: Why?

File: sdk/vault-client/tsconfig.test.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "noEmit": true
+    "noEmit": true,
```
Comment (nirhaas) [inline sdk/vault-client/tsconfig.test.json:4]: Is that ok with tsconfig? it breaks package.json iirc

File: sdk/testcontainers-vault/tsconfig.build.json [hunk 3]
```diff
@@ -1,7 +1,9 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "dist"
+    "outDir": "dist",
+    "experimentalDecorators": false,
```
Comment (nirhaas) [inline sdk/testcontainers-vault/tsconfig.build.json:5]: Why mention explicitly? same question for emitDecoratorMetadata

File: sdk/vault-client/test/collections.spec.ts [hunk 4]
```diff
@@ -1,18 +1,8 @@
 import {expect} from "chai";
 import {AsyncFunc, Context} from "mocha";
-import {Collection} from "../";
-
-
-it('coll', async function () {
```
Comment (nirhaas) [inline sdk/vault-client/test/collections.spec.ts:6]: What was that? why did you delete it?

**Review by nirhaas — APPROVED** [review]
Left some questions

File: sdk/vault-client/tsconfig.test.json (hunk 2 above)
Comment (ori-shalom) [reply sdk/vault-client/tsconfig.test.json:4]: tsconfig allow comments and trailing commas

File: sdk/vault-client/test/collections.spec.ts (hunk 4 above)
Comment (ori-shalom) [reply sdk/vault-client/test/collections.spec.ts:6]: It's not a test.
It was a small code I used to generate the baseline for the README but I modified the README manually since then so it's not relevant anymore.

File: sdk/testcontainers-vault/tsconfig.build.json (hunk 3 above)
Comment (ori-shalom) [reply sdk/testcontainers-vault/tsconfig.build.json:5]: I notice that it affect the bundle size. Typescript adds extra helpers when these are on.
We globally want it to be turned on since IDEs used the root tsconfig but when we build we turn it off for modules that don't use it.

File: tsconfig.json (hunk 1 above)
Comment (ori-shalom) [reply tsconfig.json]: Reverted
